File: 2500/solution.go
package _2500

import "sort"

func deleteGreatestValue(grid [][]int) int {
	m := len(grid)
	n := len(grid[0])
	res := 0
	max := func(a, b int) int {
		if a > b {
			return a
		}
		return b
	}
	for i := 0; i < m; i++ {
		sort.Slice(grid[i], func(a, b int) bool {
			return grid[i][a] > grid[i][b]
		})
	}
	for i := 0; i < n; i++ {
		tmp := grid[0][i]
		for j := 1; j < m; j++ {
			tmp = max(tmp, grid[j][i])
		}
		res += tmp
	}
	return res
}

File: 2500/solution.py
from typing import List


class Solution:
    def deleteGreatestValue(self, grid: List[List[int]]) -> int:
        m = len(grid)
        for i in range(m):
            grid[i].sort()
        n = len(grid[0])
        res = 0
        for i in range(n):
            tmp = grid[0][i]
            for j in range(1, m):
                tmp = max(tmp, grid[j][i])
            res += tmp
        return res
